mobile_verifier/src/speedtests.rs

```diff
@@ -1,55 +1,41 @@
+use crate::speedtests_average::{SpeedtestAverage, SPEEDTEST_LAPSE};
 use chrono::{DateTime, Duration, Utc};
 use file_store::{
     file_info_poller::FileInfoStream,
-    file_sink::{self, FileSinkClient},
+    file_sink::FileSinkClient,
     speedtest::{CellSpeedtest, CellSpeedtestIngestReport},
-    traits::TimestampEncode,
 };
 use futures::{
-    stream::{Stream, StreamExt, TryStreamExt},
+    stream::{StreamExt, TryStreamExt},
     TryFutureExt,
 };
 use helium_crypto::PublicKeyBinary;
-use helium_proto::services::poc_mobile as proto;
-use mobile_config::{client::ClientError, gateway_info::GatewayInfoResolver, GatewayClient};
-use rust_decimal::Decimal;
-use rust_decimal_macros::dec;
-use sqlx::{
-    postgres::{types::PgHasArrayType, PgTypeInfo},
-    FromRow, Postgres, Transaction, Type,
-};
-use std::{
-    collections::{HashMap, VecDeque},
-    pin::pin,
-};
+use mobile_config::{gateway_info::GatewayInfoResolver, GatewayClient};
+use sqlx::{postgres::PgRow, FromRow, Postgres, Row, Transaction};
+use std::collections::HashMap;
 use tokio::sync::mpsc::Receiver;
 
 const SPEEDTEST_AVG_MAX_DATA_POINTS: usize = 6;
-const SPEEDTEST_LAPSE: i64 = 48;
 
-#[derive(Debug, Clone, Type)]
-#[sqlx(type_name = "speedtest")]
-pub struct Speedtest {
-    pub timestamp: DateTime<Utc>,
-    pub upload_speed: i64,
-    pub download_speed: i64,
-    pub latency: i32,
-}
+pub type EpochSpeedTests = HashMap<PublicKeyBinary, Vec<Speedtest>>;
 
-impl Speedtest {
-    #[cfg(test)]
-    pub fn new(
-        timestamp: DateTime<Utc>,
-        upload_speed: i64,
-        download_speed: i64,
-        latency: i32,
-    ) -> Self {
-        Self {
-            timestamp,
-            upload_speed,
-            download_speed,
-            latency,
-        }
+#[derive(Debug, Clone)]
+pub struct Speedtest {
+    pub report: CellSpeedtest,
+}
+
+impl FromRow<'_, PgRow> for Speedtest {
```

Comment:
Not sure if file_store has sqlx already defined but you should be able to derive the FromRow against the CellSpeedtest and then your queries to grab Speedtest structs just query_as CellSpeedtest and map the results to set that result as the value of the report field in the Speedtest struct. The only real data conversion I see here is the integer types and deriving FromRow allows you to annotate integer fields to `try_from` to convert from signed to unsigned integer types